src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.ClientCertProvider.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+
+namespace System.Net.Http
+{
+    internal partial class CurlHandler : HttpMessageHandler
+    {
+
+        private sealed class ClientCertProvider : IDisposable
+        {
+            internal readonly string _certificateFile;
+            internal readonly string _password;
```

Comment:
Instead of persisting both the file and the password (even in the fallback case) try for an approach where the password is generated, used, passed to curl, and forgotten; and only the path to the file to clean up remains.